lib/std/debug.zig

```diff
@@ -655,7 +655,7 @@ pub const StackIterator = struct {
     pub fn initWithContext(first_address: ?usize, debug_info: *SelfInfo, context: *posix.ucontext_t) !StackIterator {
         // The implementation of DWARF unwinding on aarch64-macos is not complete. However, Apple mandates that
         // the frame pointer register is always used, so on this platform we can safely use the FP-based unwinder.
-        if (builtin.target.isDarwin() and native_arch == .aarch64)
+        if (builtin.target.isDarwin() and native_arch == .aarch64 and builtin.abi != .ilp32)
             return init(first_address, context.mcontext.ss.fp);
```

Comment:
This needs a comment explaining why the ABI check is necessary - it's not obvious to me, at least.

Reply:
@alexrp I wasn't sure if the comment above 

> Apple mandates that the frame pointer register is always used

was also true for this ABI check.

Also there's type collision: `fp` is defined to be `u64` but the `init()` accepts a `size`, which on ilp32 is `u32`. I'll get that reworked and/or add proper comments.